src/array.jl

```diff
@@ -212,6 +212,13 @@ end
 
             convert($A{T, N, R}, A)
         end
+
+        function vcat(A::$A...)
+            L, O = sortedmerge(map(levels, A)...)
```

Comment:
Upper case letters generally denote types. Better use `newlevels` and `isordered` here.